client/asset/eth/nodeclient.go

```diff
@@ -230,7 +240,6 @@ func (n *nodeClient) sendTransaction(ctx context.Context, txOpts *bind.TransactO
 		Value:     txOpts.Value,
 		Data:      data,
 	}), n.chainID)
-
```

Comment:
Not an issue with your diff, but this got me looking at `sendTransaction`.  I do not think this method should be using `GetPoolNonce`.  The `txOpts` provided has the nonce set.  Back in `(*ETHWallet).sendToAddr` it handles the call to `(*nodeClient).txOpts` to get and set the nonce with `nonceSendMtx` held until send is done as required.  Letting another method also use `GetPoolNonce` seems risky.

I'm also a little confused by the use of `SignTx` here because I thought we had `txOpts.Signer` set.  Does the configured `Signer` only get used in contract interactions, not regular sends?  I guess that's it.

Anyway, this is what I'm trying to say:

```diff
diff --git a/client/asset/eth/nodeclient.go b/client/asset/eth/nodeclient.go
index 656a941b..c80a3092 100644
--- a/client/asset/eth/nodeclient.go
+++ b/client/asset/eth/nodeclient.go
@@ -221,19 +221,14 @@ func (n *nodeClient) getConfirmedNonce(ctx context.Context, blockNumber int64) (
        return state.GetNonce(n.address()), nil
 }
 
-// sendTransaction sends a tx.
+// sendTransaction sends a tx. The nonce should be set in txOpts.
 func (n *nodeClient) sendTransaction(ctx context.Context, txOpts *bind.TransactOpts,
        to common.Address, data []byte) (*types.Transaction, error) {
 
-       nonce, err := n.leth.ApiBackend.GetPoolNonce(ctx, n.creds.addr)
-       if err != nil {
-               return nil, fmt.Errorf("error getting nonce: %v", err)
-       }
-
        tx, err := n.creds.ks.SignTx(*n.creds.acct, types.NewTx(&types.DynamicFeeTx{
                To:        &to,
                ChainID:   n.chainID,
-               Nonce:     nonce,
+               Nonce:     txOpts.Nonce.Uint64(),
                Gas:       txOpts.GasLimit,
                GasFeeCap: txOpts.GasFeeCap,
                GasTipCap: txOpts.GasTipCap,
@@ -276,11 +271,10 @@ func (n *nodeClient) signData(data []byte) (sig, pubKey []byte, err error) {
        return
 }
 
-func (n *nodeClient) addSignerToOpts(txOpts *bind.TransactOpts) error {
+func (n *nodeClient) addSignerToOpts(txOpts *bind.TransactOpts) {
        txOpts.Signer = func(addr common.Address, tx *types.Transaction) (*types.Transaction, error) {
                return n.creds.wallet.SignTx(*n.creds.acct, tx, n.chainID)
        }
-       return nil
 }
 
 // currentFees gets the baseFee and tipCap for the next block.
@@ -324,7 +318,7 @@ func (n *nodeClient) getCodeAt(ctx context.Context, contractAddr common.Address)
 // added automatically.
 //
 // NOTE: The nonce included in the txOpts must be sent before txOpts is used
-// again. The caller should ensure that txOpts -> send sequence is sychronized.
+// again. The caller should ensure that txOpts -> send sequence is synchronized.
 func (n *nodeClient) txOpts(ctx context.Context, val, maxGas uint64, maxFeeRate *big.Int) (*bind.TransactOpts, error) {
        baseFee, gasTipCap, err := n.currentFees(ctx)
        if err != nil {
@@ -335,20 +329,14 @@ func (n *nodeClient) txOpts(ctx context.Context, val, maxGas uint64, maxFeeRate
                maxFeeRate = new(big.Int).Mul(baseFee, big.NewInt(2))
        }
 
-       txOpts := newTxOpts(ctx, n.creds.addr, val, maxGas, maxFeeRate, gasTipCap)
-       if err := n.addSignerToOpts(txOpts); err != nil {
-               return nil, fmt.Errorf("addSignerToOpts error: %w", err)
-       }
-
        nonce, err := n.leth.ApiBackend.GetPoolNonce(ctx, n.creds.addr)
        if err != nil {
                return nil, fmt.Errorf("error getting nonce: %v", err)
        }
-       txOpts.Nonce = new(big.Int).SetUint64(nonce)
 
-       if err := n.addSignerToOpts(txOpts); err != nil {
-               return nil, fmt.Errorf("addSignerToOpts error: %w", err)
-       }
+       txOpts := newTxOpts(ctx, n.creds.addr, val, maxGas, maxFeeRate, gasTipCap)
+       txOpts.Nonce = new(big.Int).SetUint64(nonce)
+       n.addSignerToOpts(txOpts) // set txOpts.Signer
 
        return txOpts, nil
 }
```